apt/rules.bzl

```diff
@@ -235,3 +237,22 @@ deploy_apt = rule(
     Select deployment to `snapshot` or `release` repository with `bazel run //:some-deploy-apt -- [snapshot|release]
     """
 )
+
+def deploy_apt(name, target, snapshot, release, **kwargs):
```

Comment:
This should be able to either deploy to cloudsmith apt (if we use the `cloudsmith://...` convention you established) or to the previous version using curl.

Same applies to Artifact and anything else using `cloudsmith://...` endpoint.

The rules should still able to interchange with snapshot (say Docker) and release (DockerHub) right?